seacrowd/sea_datasets/flores200/flores200.py

```diff
@@ -0,0 +1,478 @@
+                        second_language_name=second_lang_name,
+                    )
+                )
+
+            BUILDER_CONFIGS.extend(seacrowd_schema_config)
+
+    DEFAULT_CONFIG_NAME = f"{_DATASETNAME}_{_LANGUAGE_NAMES[0]}_{_LANGUAGE_NAMES[1]}_source"
+
+    def _info(self) -> datasets.DatasetInfo:
+
+        if self.config.schema == "source":
+            features = datasets.Features(
+                {
+                    "id": datasets.Value("int32"),
+                    "URL": datasets.Value("string"),
+                    "domain": datasets.Value("string"),
+                    "topic": datasets.Value("string"),
+                    "has_image": datasets.Value("int32"),
+                    "has_hyperlink": datasets.Value("int32"),
+                }
+            )
+
+            features[self.config.first_language_name] = datasets.Value("string")
+            features[self.config.second_language_name] = datasets.Value("string")
+
+        else:
+            schema = str(self.config.schema).lstrip(f"{_DATASETNAME}_seacrowd_").upper()
+
+            if schema in _SCHEMAS:
+                features = SCHEMA_TO_FEATURES[schema]
+
+            else:
+                raise ValueError(f"Invalid config: {self.config.name}")
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager: datasets.DownloadManager) -> List[datasets.SplitGenerator]:
+        """Returns SplitGenerators."""
+
+        dl_dir = dl_manager.download(_URLS[_DATASETNAME])
+
+        base_dir = os.path.join(os.path.dirname(dl_dir), "flores200extracted")
+        password = "multilingual machine translation"
+
+        with zipfile.ZipFile(dl_dir, "r") as zip_ref:
+            # Set the password to extract the contents
+            zip_ref.setpassword(bytes(password, "utf-8"))
+
+            # Extract all contents to the specified directory
+            zip_ref.extractall(base_dir)
+
+        def _get_sentence_paths(split):
+            sentence_paths = [os.path.join(base_dir, _SENTENCES_PATHS[lang][split]) for lang in _LANGUAGE_NAMES]
+            return sentence_paths
+
+        return [
+            datasets.SplitGenerator(
+                name=split,
+                gen_kwargs={
+                    "sentence_paths": _get_sentence_paths(split),
+                    "metadata_path": os.path.join(base_dir, _METADATA_PATHS[split]),
+                },
+            )
+            for split in _SPLITS
+        ]
+
+    def _generate_examples(self, sentence_paths: list[str], metadata_path: str) -> Tuple[int, Dict]:
+        """Yields examples as (key, example) tuples."""
+
+        sentences = {}
+        langs = [self.config.first_language_name, self.config.second_language_name]
+
+        for path, lang in zip(sentence_paths, langs):
```

Comment:
The `sentence_path` provides the paths to the data for all available languages in Flores, while the `langs` is only lang1 and lang2, this causes misalignment between the data and the language code as shown below:

zsm_Latn - ind_Latn
```
>>> x = datasets.load_dataset('seacrowd/sea_datasets/flores200/flores200.py', name='flores200_zsm_Latn_ind_Latn_seacrowd_t2t_seacrowd_t2t')
>>> x['dev'][0]
{'id': '1', 'text_1': 'يق أورو سنين، اوق علمون دري فکولتس کدوکترن يونيۏرسيتس ستانفورد ݢڤعموم اکن جتومى الت دياݢنوستيک بارو ڽڠ جوت ݢڤيليه اتو سيل منوروت جنيهجيه: چيڤ اوبيت ڽڠ جوت ݢچيتق ڽڠ جوت ݢڤݢت دڠون ڤرينتر ايڠ\u200cکجيت ستندر دڠون يوم موڠکين نا سابوه سين ا.س. لم تيڤ سابوه.', 'text_2': 'Bak uroe Senin, awak ilmuwan dari Fakultas Kedokteran Universitas Stanford geu peuumum akan jiteumèe alat diagnostik baro nyang jeuet geupileh atoe sel meunurot jeunèhjih: chip ubit nyang jeuet geucetak nyang jeuet geupeuget deungôn printer inkjèt standar deungôn yum mungken na sabôh sen AS lam tiep sabôh.', 'text_1_name': 'zsm_Latn', 'text_2_name': 'ind_Latn'}
```

eng_Latn - ind_Latn
```
>>> x = datasets.load_dataset('seacrowd/sea_datasets/flores200/flores200.py', name='flores200_eng_Latn_ind_Latn_seacrowd_t2t_seacrowd_t2t')
>>> x['dev'][0]
{'id': '1', 'text_1': 'يق أورو سنين، اوق علمون دري فکولتس کدوکترن يونيۏرسيتس ستانفورد ݢڤعموم اکن جتومى الت دياݢنوستيک بارو ڽڠ جوت ݢڤيليه اتو سيل منوروت جنيهجيه: چيڤ اوبيت ڽڠ جوت ݢچيتق ڽڠ جوت ݢڤݢت دڠون ڤرينتر ايڠ\u200cکجيت ستندر دڠون يوم موڠکين نا سابوه سين ا.س. لم تيڤ سابوه.', 'text_2': 'Bak uroe Senin, awak ilmuwan dari Fakultas Kedokteran Universitas Stanford geu peuumum akan jiteumèe alat diagnostik baro nyang jeuet geupileh atoe sel meunurot jeunèhjih: chip ubit nyang jeuet geucetak nyang jeuet geupeuget deungôn printer inkjèt standar deungôn yum mungken na sabôh sen AS lam tiep sabôh.', 'text_1_name': 'eng_Latn', 'text_2_name': 'ind_Latn'}
```